JavaScript/JavaScript1/Week3/cactusIO.js

```diff
@@ -0,0 +1,59 @@
+// Smart phone usage app
+const activities = [];
+const usageLimit = 100; // Usage limit
+// Extra functionality for setting goals and following the streak
+const usageGoal = 60;
+let streak = 0;
+
+// Function to add an activity with automatic date
+function addActivity(activity, duration, date = new Date().toLocaleDateString()) {
+  activities.push({ date, activity, duration });
+}
+
+// Function to show activities and their total duration for today
+function showStatus(date = new Date().toLocaleDateString()) {
+  const todayActivities = activities.filter(activity => activity.date === date);
+
+  // Checking if activities have been logged first
+  if (todayActivities.length === 0) {
+    console.log("Add some activities before calling showStatus");
+    return;
+  }
+
+  let totalDuration = todayActivities.reduce((total, activity) => total + activity.duration, 0);
+  console.log(`You have added ${todayActivities.length} activities today. They amount to ${totalDuration} min. of usage`);
+
+
+  if (totalDuration > usageLimit) {
+    console.log("You have reached your limit, no more smartphoning for you!");
+  } 
+  
+  if (totalDuration <= usageGoal) {
+    streak++;
+    console.log(`Great job! You're on a ${streak}-day streak of staying within your goal.`);
+  } else {
+    streak = 0;
+    console.log("You broke your streak. Try to stay within your goal tomorrow!");
+  }
+}
+
+// Function to find the activity with the longest duration
+function mostTimeSpent() {
+  if (activities.length === 0) {
+    console.log("Add some activities before calling mostTimeSpent");
+    return;
+  }
+
+  const { activity, duration } = activities.reduce((max, activity) => (activity.duration > max.duration ? activity : max), activities[0]);
```

Comment:
I would have liked to see a comment explaining this code, instead of some of the more simple things. I understand it as a more advanced approach to doing the same thing as if you would do a `for loop` with `if else` to check which is the longest activity.